src/tests/bptree.test.ts

```diff
@@ -1,49 +1,110 @@
+	ReferencedValue,
+	binarySearchReferencedValues,
+} from "../btree/bptree";
 import { MemoryPointer } from "../btree/node";
+import { IndexFileV1 } from "../index-file/index-file";
 import { RangeResolver } from "../resolver";
+import { readBinaryFile } from "./test-util";
 
-function textEncode(phr: string): Uint8Array {
-  const encoder = new TextEncoder();
+class testMetaPage implements MetaPage {
+	private rootMP: MemoryPointer;
 
-  return encoder.encode(phr);
+	constructor(mp: MemoryPointer) {
+		this.rootMP = mp;
+	}
+
+	async root(): Promise<MemoryPointer | null> {
+		return this.rootMP;
+	}
 }
 
-class TestMetaPage implements MetaPage {
-  private _root: MemoryPointer;
+describe("test compare bytes", () => {
+	let mockRangeResolver: RangeResolver;
+	let indexFileSize: number;
+	let mockDataFileResolver: RangeResolver;
 
-  constructor(initialRoot: MemoryPointer) {
-    this._root = initialRoot;
-  }
+	beforeEach(() => {
+		mockRangeResolver = async ({ start, end }) => {
+			const indexFile = await readBinaryFile("bptree_1.bin");
 
-  setRoot(mp: MemoryPointer): void {
-    this._root = mp;
-  }
+			const slicedPart = indexFile.slice(start, end + 1);
 
-  root(): MemoryPointer {
-    return this._root;
-  }
-}
+			const arrayBuffer = slicedPart.buffer.slice(
+				slicedPart.byteOffset,
+				slicedPart.byteOffset + slicedPart.byteLength
+			);
 
-describe("test compare bytes", () => {
-  let buffer: Uint8Array;
-  let resolver: RangeResolver;
-
-  beforeEach(() => {
-    buffer = new Uint8Array(4096);
-  });
-
-  it("generates an empty tree", async () => {
-    /*
-		let mp: MemoryPointer = {
-			offset: BigInt(0),
-			length: 0,
+			return {
+				data: arrayBuffer,
+				totalLength: arrayBuffer.byteLength,
+			};
 		};
 
-		const tree = new BPTree(resolver, new TestMetaPage(mp), buffer);
+		mockDataFileResolver = async ({ start, end }) => {
+			return {
+				data: new ArrayBuffer(0),
+				totalLength: 0,
+			};
+		};
+	});
+
+	it("should read a bptree", async () => {
+		const page = new testMetaPage({ offset: 0n, length: 1 });
+		const bptree = new BPTree(mockRangeResolver, page, mockDataFileResolver);
+
+		const textEncoder = new TextEncoder();
+		const helloBuffer = textEncoder.encode("hello");
+		const key = new ReferencedValue(
+			{ offset: 0n, length: 1 },
+			helloBuffer.buffer
+		);
+
+		//const [rv, mp] = await bptree.find(key);
+
+		//console.log(rv, mp);
+	});
+
+	it("should compare reference values", () => {
+		const values: ReferencedValue[] = [
+			new ReferencedValue({ offset: 0n, length: 10 }, new Uint8Array([0])),
+			new ReferencedValue({ offset: 10n, length: 20 }, new Uint8Array([1])),
+			new ReferencedValue({ offset: 20n, length: 30 }, new Uint8Array([2])),
+		];
+
+		const key1: ReferencedValue = new ReferencedValue(
+			{ offset: 0n, length: 0 },
+			new Uint8Array([1])
+		);
+
+		const [index1, found1] = binarySearchReferencedValues(values, key1);
+
+		expect(index1).toEqual(1);
+		expect(found1).toBeFalsy();
```

Comment:
Just want to note that even though it successfully finds the index1, it still returns false.